internal/controller/controllers/conditions.go

```diff
@@ -0,0 +1,93 @@
+
+	ClusterReadyForInstallationCondition hivev1.ClusterDeploymentConditionType = "ReadyForInstallation"
+	ClusterReadyReason                   string                                = "ClusterIsReady"
+	ClusterReadyMsg                      string                                = "The cluster is ready to begin the installation"
+	ClusterNotReadyReason                string                                = "ClusterNotReady"
+	ClusterNotReadyMsg                   string                                = "The cluster is not ready to begin the installation"
+	ClusterAlreadyInstallingReason       string                                = "ClusterAlreadyInstalling"
+	ClusterAlreadyInstallingMsg          string                                = "The cluster cannot begin the installation because it has already started"
+
+	ClusterValidatedCondition    hivev1.ClusterDeploymentConditionType = "Validated"
+	ClusterValidationsOKMsg      string                                = "The cluster's validations are passing"
+	ClusterValidationsUnknownMsg string                                = "The cluster's validations have not yet been calculated"
+	ClusterValidationsFailingMsg string                                = "The cluster's validations are failing:"
+
+	//Agent Conditions
+	SpecSyncedCondition conditionsv1.ConditionType = "SpecSynced"
+
+	ConnectedCondition      conditionsv1.ConditionType = "Connected"
+	AgentConnectedReason    string                     = "AgentIsConnected"
+	AgentDisconnectedReason string                     = "AgentIsDisconnected"
+	AgentConnectedMsg       string                     = "The agent's connection to the installation service is unimpaired"
+	AgentDisonnectedMsg     string                     = "The agent has not contacted the installation service in some time, user action should be taken"
+
+	InstalledCondition conditionsv1.ConditionType = "Installed"
+
+	ReadyForInstallationCondition conditionsv1.ConditionType = "ReadyForInstallation"
+	AgentReadyReason              string                     = "AgentIsReady"
+	AgentReadyMsg                 string                     = "The agent is ready to begin the installation"
+	AgentNotReadyReason           string                     = "AgentNotReady"
+	AgentNotReadyMsg              string                     = "The agent is not ready to begin the installation"
+	AgentAlreadyInstallingReason  string                     = "AgentAlreadyInstalling"
+	AgentAlreadyInstallingMsg     string                     = "The agent cannot begin the installation because it has already started"
+
+	ValidatedCondition         conditionsv1.ConditionType = "Validated"
+	AgentValidationsPassingMsg string                     = "The agent's validations are passing"
+	AgentValidationsUnknownMsg string                     = "The agent's validations have not yet been calculated"
+	AgentValidationsFailingMsg string                     = "The agent's validations are failing:"
```

Comment:
Do you have an equivalent for "Stopped" in https://github.com/openshift/hive/pull/1356, meaning we are no longer processing work on this cluster. (it implies neither success nor failure, must combine with Completed / Failed to know)

Reply:
Last one we had was Failed, meaning an attempt failed (but we might still be retrying). 

Talked to @rollandf on slack though and we can look into all these in a future PR.

Reply:
installation failure will be updated according to https://issues.redhat.com/browse/MGMT-4296

Reply:
If something else is required then please let us know.

Reply:
cc: @RazRegev